### ISSUE TITLE
 jsx-no-leaked-render removes parentheses effecting formatting

### ISSUE THREAD

**Issue by Jackman3005** [jsx-eslint/eslint-plugin-react#3357]
Hello I am trying to get the jsx-no-leaked-render to fix some code in my repo. When it runs its fix it is changing more code than necessary. In this case, it is removing parentheses around the element. Although, not strictly necessary, it's a convention we use in our codebase to keep formatting consistent resulting in more clearly scoped conditional statements and components.

Although one could argue it doesn't matter for some people or its not a big deal, I think it's best that a rule not have any unintended effect on code structure other than what it is built to fix.

Note: This also happens when fixing strategy is set to coerce.

# Before fixing
```jsx
{item.validateResult && item.validateResult.error && (
  <ErrorWrapper>
    <ErrorIconContainer>
      <ErrorIcon name="info" size={40} />
    </ErrorIconContainer>
    <Text size="medium" $warning>
      {item.validateResult.error.message}
    </Text>
  </ErrorWrapper>
)}
```

# After fixing
```jsx
{item.validateResult && item.validateResult.error ? <ErrorWrapper>
    <ErrorIconContainer>
      <ErrorIcon name="info" size={40} />
    </ErrorIconContainer>
    <Text size="medium" $warning>
      {item.validateResult.error.message}
    </Text>
  </ErrorWrapper> : null}
```

**Comment by ljharb** [comment]
I agree, it shouldn't be removing these parens.

cc @Belco90

**Comment by Jackman3005** [comment]
I wouldn't mind helping with a PR here, but I think I need someone to explain what they think the problem is and where I'd have to make a fix. Happy to take a stab at figuring things out after that.

Maybe @ljharb  or @Belco90 could point me in the right direction?

Thanks in advance!

**Comment by ljharb** [comment]
My personal approach would likely be to add some valid and invalid test cases, run `npm run mocha path/to/test/files`, and then in the rule file, console.log → rerun → debug → edit → repeat :-)

**Comment by ljharb** [comment]
@Jackman3005 have you had a chance to take a look at this?

**Comment by Jackman3005** [comment]
Hey Jordan, no I have not yet, sorry. I got caught up in other things and have left this rule out from my linting for now.

**Comment by Jackman3005** [comment]
Also, I saw someone else mention a similar(?) [lint rule that works with Typescript](https://www.npmjs.com/package/eslint-plugin-jsx-expressions) which would solve issues where this rule was adding `!!` to boolean variables (I'm lead to believe at least). I haven't checked it out yet, but I will probably investigate that before coming back and working on this.
